channels/rdpear/client/rdpear_main.c

```diff
@@ -0,0 +1,965 @@
+/**
+ * FreeRDP: A Remote Desktop Protocol Implementation
+ * Authentication redirection virtual channel
+ *
+ * Copyright 2023 David Fort <contact@hardening-consulting.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <krb5/krb5.h>
+#include <errno.h>
+
+#include <winpr/assert.h>
+
+#include <winpr/crt.h>
+#include <winpr/wlog.h>
+#include <winpr/print.h>
+#include <winpr/asn1.h>
+#include <winpr/sspi.h>
+#include <winpr/collections.h>
+
+#include <rdpear-common/ndr.h>
+#include <rdpear-common/rdpear_common.h>
+#include <rdpear-common/rdpear_asn1.h>
+
+#include <freerdp/config.h>
+#include <freerdp/freerdp.h>
+#include <freerdp/addin.h>
+#include <freerdp/client/channels.h>
+#include <freerdp/channels/log.h>
+#include <freerdp/channels/rdpear.h>
+
+#define TAG CHANNELS_TAG("rdpear.client")
+
+/* defined in libkrb5 */
+krb5_error_code encode_krb5_authenticator(const krb5_authenticator* rep, krb5_data** code_out);
```

Comment:
did report this upstream so these may be exposed officially?
or is there a way to reproduce this in a more stable manner (stuff not part of headers tends to change over time)

Reply:
These functions are kinda automatically generated with macros in libkrb5 (based on message definition) and so I think all the marshalling functions for all kerberos messages are exported even if not present in `krb5.h` (so that also include marshalling every pieces of a message). So I'm rather confident that this would not move that much as libkrb5 modules may also need to forge some of these messages.
As it's rather painful to reimplement these, I'd be for relying on the exported functions from libkr5 and we'll implement our own if it happens that it's needed for some platforms.

Reply:
@hardening maybe write a mail upstream to clarify this? (did receive quite the interesting response about our use of the profile API)

Reply:
@akallabeth I will